File: Bybliotheca/src/main/java/pi/bybliotheca/entity/Book.java
package pi.bybliotheca.entity;


import javax.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@Entity
@Table(name = "BOOKS")

public class Book {
    @Id
    @GeneratedValue
    private int id;
    private String title;
    private String author;
    private String summary;
    private String genre;
    private int npages;
    private int year;
    private int qty;

    private String img;

    private List<Long> reviews;

    private List<Integer> ratings;

    private double avrRating;


    public Book(String title, String author, String summary, String genre, int npages, int year, int qty) {
        this.title = title;
        this.author = author;
        this.summary = summary;
        this.genre = genre;
        this.npages = npages;
        this.year = year;
        this.qty = qty;
    }

    public double calculateAvrRating(){
        int ratingSum = this.ratings.stream()
                .mapToInt(Integer::intValue)
                .sum();
        double avrRating = (double) ratingSum/this.ratings.size();
        return avrRating;

    }
}

File: bybliotheca_flutter/lib/entities/book.dart
class Book {
  int id;
  String title;
  String author;
  String summary;
  String genre;
  int npages;
  int year;
  int qty;

  Book({
    this.id,
    this.title,
    this.author,
    this.summary,
    this.genre,
    this.npages,
    this.year,
    this.qty,
});

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['id'],
      title: json['title'],
      author: json['author'],
      summary: json['summary'],
      genre: json['genre'],
      npages: json['npages'],
      year: json['year'],
      qty: json['qty'],
    );
  }
  
}